File: satwatch_v2.0/app/src/main/java/com/sdk/satwatch/util/RxUtil.java
package com.sdk.satwatch.util;

import android.os.Looper;

import io.reactivex.CompletableTransformer;
import io.reactivex.FlowableTransformer;
import io.reactivex.ObservableTransformer;
import io.reactivex.SingleTransformer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class RxUtil {

    public static <T> ObservableTransformer<T, T> IoToMainObserve() {
        return upstream -> {
            if (Looper.myLooper() != null || Looper.myLooper() == Looper.getMainLooper()) {
                return upstream.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread());
            } else {
                return upstream.observeOn(AndroidSchedulers.mainThread());
            }
        };
    }

    public static <T> SingleTransformer<T, T> IoToMainSingle() {
        return upstream -> {
            if (Looper.myLooper() != null || Looper.myLooper() == Looper.getMainLooper()) {
                return upstream.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread());
            } else {
                return upstream.observeOn(AndroidSchedulers.mainThread());
            }
        };
    }

    public static CompletableTransformer IoToMainComplete() {
        return upstream -> {
            if (Looper.myLooper() != null || Looper.myLooper() == Looper.getMainLooper()) {
                return upstream.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread());
            } else {
                return upstream.observeOn(AndroidSchedulers.mainThread());
            }
        };
    }

}

File: rvigor/app/src/main/java/com/healthy/rvigor/util/RxUtil.kt
package com.healthy.rvigor.util

import android.os.Looper
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.ObservableTransformer
import io.reactivex.rxjava3.schedulers.Schedulers

/**
 * @Description:    Rvigor
 * @Author:         wb
 * @CreateDate:     2024/5/6 9:49
 * @UpdateRemark:
 */
object RxUtil {

    fun IoToMainObserve(): ObservableTransformer<Any, Any>? {
        return ObservableTransformer<Any, Any> { upstream ->
            if (Looper.myLooper() != null || Looper.myLooper() == Looper.getMainLooper()) {
                return@ObservableTransformer upstream.subscribeOn(Schedulers.io())
                    .observeOn(AndroidSchedulers.mainThread())
            } else {
                return@ObservableTransformer upstream.observeOn(AndroidSchedulers.mainThread())
            }
        }
    }


}